### DISCUSSION TITLE
unable to update

### DISCUSSION THREAD

**Discussion by aboudsawan** [MiKTeX/miktex#1461] [Q&A]
Hello,
Tried to install MikTex on Windows 11, but I got the following error "ctan.yazd.ac.ir took too long to respond"
So I searched for the installation file "basic-miktex-24.1-x64.exe" on Google and found another source to install MikTex.

Now when I do update, I get the following:
Timeout was reached

Detail:
code="28", url="https://ctan.yazd.ac.ir/systems/win32/miktex/tm/packages/miktex-zzdb3-2.9.tar.lzma".

please provide help on this issue... why am I being directed to ""ctan.yazd.ac.ir" for installation or update?

Thnx

ps: geographically, I am close to Iran, do I need to do something manually to alter this installation/update setup?

**Comment by aboudsawan** [comment]
I downloaded the MiKTeX Setup Utility "miktexsetup-5.5.0+1763023-x64.zip"... and as before I have to google it since the download link in mikttex.org is not working.
I used this utility to download MiKTeX into a local package repository from a **specific remote repository**:
for example ... as follows;

miktexsetup_standalone --verbose --local-package-repository=C:\miktex-repository --remote-package-repository=https://tug.ctan.org/systems/win32/miktex/tm/packages/ --package-set=complete download

**Comment by nick0413** [comment]
I am getting the same error, but only in one computer, it is very weird. I cant find anything about it. Im on windows 11 and it works fine in my win10 PC.

**Reply by husayt** [reply]
i get the same error and can't change repository. it only offers random repo, but can't see way to point to specific repo

**Reply by aglobalphenomenon** [reply]
Please check the answer below, hope it helps.

**Comment by aglobalphenomenon** [comment]
I know this is late, but in case someone faces the same issue, I managed to stitch a simple solution in my case (on Windows).

When updating packages, MiKTeX’s package manager (`mpm`) automatically selects the closest mirror. Sometimes the closest mirror (e.g. one at https://ctan.yazd.ac.ir/) may be hosted in a region (like Iran) where domains are blocked by your region, causing connection timeouts (error code 28).

1. **Manually Set an Accessible Repository**  
   - **Find a Mirror:**  
     Visit [[MiKTeX Repositories](https://miktex.org/pkg/repositories)](https://miktex.org/pkg/repositories) and choose a URL from a country with no geo-blocking issues.
   - **Configure via Command Line:**  
     Open a command prompt and run the following (run elevated/as admin if needed):
     ```
     mpm --set-repository="https://<chosen-repository>/ctan/systems/win32/miktex/tm/packages/"
     ```
     If running in administrator mode (using MiKTeX Console Admin), add the `--admin` flag and use:
     ```
     mpm --admin --set-repository="https://<chosen-repository>/ctan/systems/win32/miktex/tm/packages/"
     ```
     Example :
     ```
     mpm --admin --set-repository="https://mirrors.nxthost.com/ctan/systems/win32/miktex/tm/packages/"
     ```

2. **Use a VPN**  
   Otherwise, instead of all of that, you can use a VPN to temporarily bypass the geo-blocking to reach a different default mirror.
